### DISCUSSION TITLE
Trying to configure renovate with private gitlab

### DISCUSSION THREAD

**Discussion by i-am-learning-still** [renovatebot/renovate#29904] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab, renovate 37.413.4

### Please tell us more about your question or problem

I have installed renovate through npm. I am trying to connect it to my private gitab repository, but renovate is not connecting to gitlab. I have set RENOVATE_SSL_KEY,RENOVATE_ENDPOINT, RENOVATE_TOKEN, RENOVATE_SSL_CERT

### Logs (if relevant)

<details><summary>Logs</summary>

```

Error: "message": "external-host-error", "stack": "Error: external-host-error\n at GitlabHttp.request (/home/.nvm/versions/node/v20.13.1/lib/node_modules/renovate/lib/util/http/gitlab.ts:82:15)\n at processTicksAndRejections (node:internal/process/task_queues:95:5)\n at GitlabHttp.requestJson FATAL: Authentication failure INFO: Renovate is exiting with a non-zero code due to the following logged errors DEBUG: GET https://gitlab.myrepo.com/user = (code=ERR_BODY_PARSE_FAILURE, statusCode=200 retryCount=0, duration=195) WARN: Retry-After: unexpected status code 200 "url": "https://myrepo.cloudflareaccess.com/test/access"
DEBUG: Gitlab API error
"err": {
"name": "ParseError",
"code": "ERR_BODY_PARSE_FAILURE", "message": "Unexpected token '<', "<!DOCTYPE "... is not valid JSON in "https://myrepo.cloudflareaccess.com/test/access/login/gitlab.myrepo.com?kid=MjBhN" "stack": "RequestError: Unexpected token '<', "<!DOCTYPE "... is not valid JSON in "message": "Unexpected token '<', "<!DOCTYPE "... is not valid JSON in "https://myrepo.cloudflareaccess.com/test/access/login/gitlab.myrepo.com?kid=N"
```

</details>

**Comment by RahulGautamSingh** [comment]
`RENOVATE_SSL_KEY`, `RENOVATE_SSL_CERT` are not valid env vars ie.renovate does not recognize or use them.

The `RENOVATE_ENDPOINT` you set seems wrong...because the request is being sent `https://gitlab.myrepo.com/user`. Repo name should not be present in the endpoint.

Use the `repositories` config option to set the repo name.
```js
module.exports = {
  token: '**gitlab_token**',
  platform: 'gitlab',
  repositories: ['username/repo', 'orgname/repo'],
}
```

Checkout the [getting-started](https://docs.renovatebot.com/getting-started/use-cases/) and [self-hosting](https://docs.renovatebot.com/examples/self-hosting/) docs.

**Comment by i-am-learning-still** [comment]
This could be an issue with mTLS, since i observe a redirection to "cloudflare". When we do a git clone, with the gitlab token only, there is redirection, if we do a git clone with gitlab token as well as the mTLS files, the clone happens. Which is why we thought of providing RENOVATE_SSL_KEY, RENOVATE_SSL_CERT parameters. So, does renovate support connecting using mTLS parameters?

**Reply by RahulGautamSingh** [reply]
Check this out: https://docs.renovatebot.com/examples/self-hosting/#self-signed-tlsssl-certificates
